python/lsst/ctrl/mpexec/cli/script/qgraph.py

```diff
@@ -0,0 +1,156 @@
+    input : `str`
+        Comma-separated names of the input collection(s). Entries may include a
+        colon (:), the first string is a dataset type name that restricts the
+        search in that collection.
+    output : `str`
+        Name of the output CHAINED collection. This may either be an existing
+        CHAINED collection to use as both input and output (if `input` is
+        `None`), or a new CHAINED collection created to include all inputs
+        (if `input` is not `None`). In both cases, the collection's children
+        will start with an output RUN collection that directly holds all new
+        datasets (see `output_run`).
+    output_run : `str`
+        Name of the new output RUN collection. If not provided then `output`
+        must be provided and a new RUN collection will be created by appending
+        a timestamp to the value passed with `output`. If this collection
+        already exists then `extend_run` must be passed.
+    extend_run : `bool`
+        Instead of creating a new RUN collection, insert datasets into either
+        the one given by `output_run` (if provided) or the first child
+        collection of `output` (which must be of type RUN).
+    replace_run : `bool`
+        Before creating a new RUN collection in an existing CHAINED collection,
+        remove the first child collection (which must be of type RUN). This can
+        be used to repeatedly write to the same (parent) collection during
+        development, but it does not delete the datasets associated with the
+        replaced run unless `prune-replaced` is also True. Requires `output`,
+        and `extend_run` must be `None`.
+    prune_replaced : "unstore", "purge", or `None`.
+        If not `None`, delete the datasets in the collection replaced by
+        `replace_run`, either just from the datastore ("unstore") or by
+        removing them and the RUN completely ("purge"). Requires `replace_run`.
+    data_query : `str`
+        User query selection expression.
+    show : `list` [`str`] or `None`
+        Descriptions of what to dump to stdout.
+
+    Returns
+    -------
+    qgraph : `lsst.pipe.base.QuantumGraph`
+        The qgraph object that was created.
+    """
+    # if pipeline is not None and qgraph is not None:
+    #     raise ClickException(
+    #         "Do not pass '--qgraph' file location if running 'build' command before 'qgraph'.")
+    # if pipeline is None and qgraph is None:
+    #     raise ClickException(  # or, could make an anonymous pipeline...?
+    #         "Run 'build' command before 'qgraph' or provide a serialzed qgraph file location.")
+
+    if log_level is not None:
+        CliLog.setLogLevels(log_level)
+
+    class MakeGraphArgs:
+        """A container class for arguments to CmdLineFwk.makeGraph, whose
+        API (currently) is written to accept inputs from argparse in a generic
+        container class.
+        """
```

Comment:
I think `types.SimpleNamespace` class would work better for this: https://docs.python.org/3/library/types.html#types.SimpleNamespace

Reply:
from the docs:
> SimpleNamespace may be useful as a replacement for class NS: pass

I had basically done that (`class NS: pass`) for the `build` subcommand, @timj did not like it and that's why I wrote the Args classes.

Reply:
What didn't I like?

Reply:
in https://github.com/lsst/ctrl_mpexec/pull/58, search for "cute" (I don't see a way to link to the particular conversation)

Reply:
`Args` was declared 
```
class Args
    pass
args = Args()
args.qgraph = qgraph
args.save_qgraph = save_qgraph
<etc>
```

Reply:
I think you are over interpreting my comment. My final quote was:

> At the very least a comment like you just made in the class description for Args would be useful.

(I haven't looked at the current context -- SimpleNamespace does look useful as an explicit declaration of intent)